extensions/src/main/scala/ai/h2o/sparkling/extensions/rest/api/ChunkServlet.scala

```diff
@@ -17,113 +17,131 @@
 
 package ai.h2o.sparkling.extensions.rest.api
 
-import java.nio.{ByteBuffer, ByteOrder}
-
 import ai.h2o.sparkling.utils.ScalaUtils._
 import ai.h2o.sparkling.utils.Base64Encoding
-import ai.h2o.sparkling.extensions.serde.{SerializationUtils, ChunkAutoBufferWriter}
+import ai.h2o.sparkling.extensions.serde.{ChunkAutoBufferReader, ChunkAutoBufferWriter, ChunkSerdeConstants}
 import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
 import water.DKV
 import water.fvec.Frame
 import water.server.ServletUtils
 
 /**
-  * This servlet class handles GET requests for the path /3/Chunk
-  * It requires 4 get parameters
-  * - frame_name - a unique string identifier of H2O Frame
-  * - chunk_id - a unique identifier of the chunk within the H2O Frame
-  * - expected_type - byte array encoded in Base64 encoding. The types corresponds to the `selected_columns` parameter
-  * - selected_columns - selected columns indices encoded int Base64 encoding.
-  * The result is represented as a stream of binary data. Data are encoded to AutoBuffer row by row.
-  * The data stream starts with the integer representing the number of rows.
+  * This servlet class handles GET and PUT requests for the path /3/Chunk
   */
 final class ChunkServlet extends HttpServlet {
 
-  private case class RequestParameters(
+  private case class GetRequestParameters(
```

Comment:
Does this class represent parameters for the `GET` request? Shouldn't the case class be named `GETRequestParameters` instead?

Reply:
Ok, no problem